### ISSUE TITLE
ResumePause doesn't always register. Play/Pause Icon is confused.

### ISSUE THREAD

**Issue by MrGibus** [aome510/spotify-player#387]
**Describe the bug**
The Icon may show play icon while it is actually paused (and vice versa) pressing space will correct the icon but not play.

**To Reproduce**
Run program -> pause shown
Press space -> music plays, icon flips from paused to play to pause again.
Press space -> icon flips to play
press space -> music pauses, icon flips from play to pause and the play again.
Press space -> icon flips to pause.

If we navigate to another song after loading the program and play that it will play initially but the icons still get confused when resuming.

**Expected behaviour**
Run program -> pause shown
Press space -> music plays, icon flips from paused to play
press space -> music pauses, icon flips from play to pause

**Log and backtrace**
The following log snippet looks related.
```
2024-03-04T05:50:11.958455Z  INFO client_request{request=Player(ResumePause)}: spotify_player::client: successfully handled the client request, took: 226ms
2024-03-04T05:50:13.989474Z  INFO client_request{request=Player(ResumePause)}: spotify_player::client: successfully handled the client request, took: 235ms
2024-03-04T05:50:14.357216Z  INFO spotify_player::streaming: Got a new player event: Paused { track_id: TrackId("6BUWdfmuPehKO5E7pznn1L"), position_ms: 55474, duration_ms: 375272 }
2024-03-04T05:50:14.591747Z  INFO client_request{request=GetCurrentPlayback}: spotify_player::client: successfully handled the client request, took: 234ms
2024-03-04T05:50:18.160894Z  INFO client_request{request=Player(ResumePause)}: spotify_player::client: successfully handled the client request, took: 287ms
2024-03-04T05:50:18.162149Z  INFO spotify_player::streaming: Got a new player event: Playing { track_id: TrackId("6BUWdfmuPehKO5E7pznn1L"), position_ms: 55474, duration_ms: 375272 }
2024-03-04T05:50:18.162361Z  INFO spotify_player::streaming: Got a new player event: Playing { track_id: TrackId("6BUWdfmuPehKO5E7pznn1L"), position_ms: 55474, duration_ms: 375272 }
2024-03-04T05:50:18.404869Z  INFO client_request{request=GetCurrentPlayback}: spotify_player::client: successfully handled the client request, took: 242ms
2024-03-04T05:50:18.457483Z  INFO client_request{request=GetCurrentPlayback}: spotify_player::client: successfully handled the client request, took: 295ms
```

**Screenshots**
Can add on request.

**Environment**

- Application version: 0.17.0
- Application features: Default.

```
             .',;::::;,'.                Gibus@Gibus
         .';:cccccccccccc:;,.            -----------
      .;cccccccccccccccccccccc;.         OS: Fedora Linux 39 (Thirty Nine) x86_64
    .:cccccccccccccccccccccccccc:.       Host: B550 AORUS PRO AX
  .;ccccccccccccc;.:dddl:.;ccccccc;.     Kernel: 6.6.13-200.fc39.x86_64
 .:ccccccccccccc;OWMKOOXMWd;ccccccc:.    Uptime: 4 hours, 9 mins
.:ccccccccccccc;KMMc;cc;xMMc:ccccccc:.   Packages: 2282 (rpm)
,cccccccccccccc;MMM.;cc;;WW::cccccccc,   Shell: bash 5.2.26
:cccccccccccccc;MMM.;cccccccccccccccc:   Resolution: 1920x1080, 2560x1440, 1920x1080
:ccccccc;oxOOOo;MMM0OOk.;cccccccccccc:   WM: i3
cccccc:0MMKxdd:;MMMkddc.;cccccccccccc;   Theme: oomox-Pandora-Arc [GTK2/3]
ccccc:XM0';cccc;MMM.;cccccccccccccccc'   Icons: Vortex-Dark-Icons [GTK2/3]
ccccc;MMo;ccccc;MMW.;ccccccccccccccc;    Terminal: WezTerm
ccccc;0MNc.ccc.xMMd:ccccccccccccccc;     CPU: AMD Ryzen 5 5600X (12) @ 3.700GHz
cccccc;dNMWXXXWM0::cccccccccccccc:,      GPU: NVIDIA GeForce RTX 3060 Ti
cccccccc;.:odl:.;cccccccccccccc:,.       Memory: 3036MiB / 32032MiB
:cccccccccccccccccccccccccccc:'.
.:cccccccccccccccccccccc:;,..
  '::cccccccccccccc::;,.
```

**Additional context**
Following may be relevant:
- Running Pipewire (assume default backend?)
- own Client ID, No other changes to app.toml
- Tried with Default keymap and same issue
- Had spotify-tui installed prior to install

**Comment by aome510** [comment]
@MrGibus thanks for reporting. Should be fixed by https://github.com/aome510/spotify-player/pull/388. Can you checkout and try that patch?

**Comment by MrGibus** [comment]
> @MrGibus thanks for reporting. Should be fixed by #388. Can you checkout and try that patch?

Wow that was quick.
Tried the patch and it's working as expected.

Thanks. :)
